node/src/proxy_server/mod.rs

```diff
@@ -1617,6 +1646,70 @@ mod tests {
         }
     }
 
```

Comment:
The following tests address `get_return_route_info()` directly (white-box tests) and make sure it operates as intended.